lib/galaxy/webapps/galaxy/config_schema.yml

```diff
@@ -589,6 +589,13 @@ mapping:
           will use a less efficient monitoring scheme that may work in wider range of scenarios than
           the watchdog default.
 
+      refgenie_config_file:
+        type: str
+        default: ''
```

Comment:
Does it *need* to be an empty string if not set? If a default is not specified in the schema, `app.config.refgenie_config_file` will have the value `None`, which would be the expected behavior, right?

Reply:
No, it should be `null` as shown before 0c86a4fc3dfba22f29141f96741386d8afb767ef, but unfortunately this does not pass tests when set.

It is set to an empty string to pass the config syntax / config tests. This is very unfortunate. The inconsistency in behavior between a "missing" `default` value and having `default: null` is really, really bad. Explicit is better than implicit, and they should definitely not act differently. Because of this behavior, while `null` makes more logical sense (and was the original default), explicitly setting the default to `''` is, as a whole, less incorrect.

Better fix is to correct the behavior of the config testing / verification to enable the default to be explicitly setable to `null`, but that is outside the scope of this PR. It could potentially be done with the addition of a `nullable` attribute, that allows values to be set as null or not.

Reply:
I've convinced myself to pretend that I didn't notice the really, really bad inconsistency (should still be addressed), and am back to `null` in 17ba3d07752bd8cfa3d39b68a73c96626599830d thanks @ic4f !

Reply:
@blankenberg Is the inconsistency due to galaxy's config handling? The config system is not without fault, but a simple matter of creating an attribute based on the schema should work correctly. The attribute is created (regardless of whether it's set or not) in this loop: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/__init__.py#L222
(So it's `null` by default.) The config integration test should work automatically too. If that didn't happen as expected, please let me know and I'll look into it - that would need to be fixed.